src/helpers/unmountRoot.ts

```diff
@@ -0,0 +1,24 @@
+import { reconciler } from '../core/reconciler';
+import { roots } from '../core/roots';
+import { store } from '../store';
+import { type Root } from '../typedefs/Root';
+
+export function unmountRoot(root: Root)
+{
+    store.unmountQueue.delete(root);
+
+    const fiber = root.fiber;
+
+    if (fiber)
+    {
+        reconciler.updateContainer(null, fiber, null, () =>
+        {
+            if (root.applicationState.app)
+            {
+                root.applicationState.app.destroy();
```

Comment:
Forgive my ignorance - I'm not very familiar with `react-reconciler`, but it strikes me that I should be able to simulate this behavior as follows in a component.

```
const MyComponent: React.FC = () => {
  const ref = useRef()
  useEffect(() => {
    return () => ref.current?.destroy()
  }, [])
  <Application ref={ref}>...</Application>
}
```

I just tried this to see if it would fix the memory leak woes, but it appears to be insufficient. Is there something about the `updateContainer` call that's also necessary to fully release the memory allocated by `<Application>`? 

I also tried this with all the `destroy` options set to true - still no luck.

Reply:
The issue isn't calling `destroy` on the Application, but rather the root being created via `createRoot` and stored in the global roots cache isn't being removed. The `updateContainer` with `null` first removes any child components, and then `delete` is called on the map to hopefully clean up the now empty fiber root.

Reply:
Gotcha - do we need to also clear `root.application.resizeTo` here? Or is it possible that if we set this to say, `window`, that we'd be creating a duplicate of `window` that persists?